src/libcore/iter/traits.rs

```diff
@@ -462,8 +462,8 @@ impl<'a, I: DoubleEndedIterator + ?Sized> DoubleEndedIterator for &'a mut I {
 /// # impl Iterator for Counter {
 /// #     type Item = usize;
 /// #     fn next(&mut self) -> Option<usize> {
-/// #         self.count += 1;
 /// #         if self.count < 6 {
+/// #             self.count += 1;
```

Comment:
Reducing the number in the conditional to `5` would keep the length the same. Keeping 5 seems nice. Needs adjustment to 5 in size_hint.